client/transport.go

```diff
@@ -21,7 +21,9 @@ func (tf transportFunc) RoundTrip(req *http.Request) (*http.Response, error) {
 func resolveTLSConfig(transport http.RoundTripper) (*tls.Config, error) {
 	switch tr := transport.(type) {
 	case *http.Transport:
-		return tr.TLSClientConfig, nil
+		if tr.DialTLS != nil {
+			return tr.TLSClientConfig, nil
+		}
 	case transportFunc:
 		return nil, nil // detect this type for testing.
 	default:
```

Comment:
Shall I add the return nil, nil after line 26 through ? I did not managed to get the "19:57:02 client/transport.go:32: missing return at end of function" error with lint or vet